From: "thomas.layh" <thomas.layh@aoe.com>
Subject: Fix deprecations in 8.3

===================================================================
diff --git a/Classes/Command/BuildQueueCommand.php b/extensions/crawler/Classes/Command/BuildQueueCommand.php
--- a/Classes/Command/BuildQueueCommand.php	
+++ b/Classes/Command/BuildQueueCommand.php	(date 1715841877782)
@@ -134,9 +134,9 @@
 
         $pageId = MathUtility::forceIntegerInRange((int) $input->getArgument('page'), 0);
         if ($pageId === 0) {
-            $message = "Page ${pageId} is not a valid page, please check you root page id and try again.";
+            $message = "Page {$pageId} is not a valid page, please check you root page id and try again.";
             MessageUtility::addErrorMessage($message);
-            $output->writeln("<info>${message}</info>");
+            $output->writeln("<info>{$message}</info>");
             return 1;
         }
