Index: extensions/crawler/Classes/Command/BuildQueueCommand.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/extensions/crawler/Classes/Command/BuildQueueCommand.php b/extensions/crawler/Classes/Command/BuildQueueCommand.php
--- a/extensions/crawler/Classes/Command/BuildQueueCommand.php	
+++ b/extensions/crawler/Classes/Command/BuildQueueCommand.php	(date 1715841877782)
@@ -134,9 +134,9 @@
 
         $pageId = MathUtility::forceIntegerInRange((int) $input->getArgument('page'), 0);
         if ($pageId === 0) {
-            $message = "Page ${pageId} is not a valid page, please check you root page id and try again.";
+            $message = "Page {$pageId} is not a valid page, please check you root page id and try again.";
             MessageUtility::addErrorMessage($message);
-            $output->writeln("<info>${message}</info>");
+            $output->writeln("<info>{$message}</info>");
             return 1;
         }
 
