diff --git a/Configuration/TypoScript/ImageRendering/setup.typoscript b/Configuration/TypoScript/ImageRendering/setup.typoscript
index 9bf8fe7..1e3adac 100755
--- a/Configuration/TypoScript/ImageRendering/setup.typoscript
+++ b/Configuration/TypoScript/ImageRendering/setup.typoscript
@@ -4,7 +4,7 @@
 # Including library for processing of magic images and file abstraction attributes on img tag
 #******************************************************

-lib.parseFunc_RTE {
+lib.parseFunc {
     tags.img = TEXT
     tags.img {
         current = 1
