diff --git a/Resources/Public/JavaScript/GridElementsDragDrop.js b/Resources/Public/JavaScript/GridElementsDragDrop.js
index 09b2122..6cf3d2b 100644
--- a/Resources/Public/JavaScript/GridElementsDragDrop.js
+++ b/Resources/Public/JavaScript/GridElementsDragDrop.js
@@ -111,7 +111,7 @@ define(['jquery', 'jquery-ui/droppable', 'TYPO3/CMS/Backend/LayoutModule/DragDro
         DragDrop.default.column = $element.closest(DragDrop.default.columnIdentifier);
         DragDrop.default.column.removeClass('active');

-        $element.parents(DragDrop.default.columnHolderIdentifier).find(DragDrop.default.addContentIdentifier).hide();
+        // $element.parents(DragDrop.default.columnHolderIdentifier).find(DragDrop.default.addContentIdentifier).hide();
         $element.find(DragDrop.default.dropZoneIdentifier).hide();

         // make the drop zones visible
