diff --git a/Classes/Helper/Helper.php b/Classes/Helper/Helper.php
index f354638..bbcf619 100644
--- a/Classes/Helper/Helper.php
+++ b/Classes/Helper/Helper.php
@@ -185,8 +185,10 @@ public function getPidFromUid(int $uid = 0): int
             )
             ->execute()
             ->fetch();
-        $pid = (int)$triggerElement['pid'];
-        return is_array($triggerElement) && $pid ? $pid : 0;
+        if ($triggerElement === false) {
+            return 0;
+        }
+        return (int)($triggerElement['pid'] ?? 0);
     }
 
     /**
